openapi-generator/templates/go/model.mustache

```diff
@@ -34,7 +34,7 @@ type {{classname}} struct {
 {{#description}}
 	// {{{description}}}
 {{/description}}
-	{{name}} {{#isNullable}}*{{/isNullable}}{{^isNullable}}{{#isBoolean}}*{{/isBoolean}}{{/isNullable}}{{{dataType}}} `json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
+	{{name}} {{#isNullable}}*{{/isNullable}}{{^isNullable}}{{#isBoolean}}*{{/isBoolean}}{{/isNullable}}{{{dataType}}} `{{#vendorExtensions.x-go-custom-tag}}{{{.}}}{{/vendorExtensions.x-go-custom-tag}}{{^vendorExtensions.x-go-custom-tag}}json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{/vendorExtensions.x-go-custom-tag}}`
```

Comment:
These templates are a bit difficult to read but it removes `omitempty` when a the custom tag is set right? If not sure how extensible the schema is but should we maybe introduce a new field like `go-allow-empty` to the schema and set `omitempty` based on that boolean value? Otherwise using `x-go-custom-tag` will now always drop `omitempty` which might not be wanted in every case

Reply:
Yes, I realize this downside. I can try to rename it and see if just making a new one works.

Reply:
Okay, different names worked too, so then I could redo it much more cleanly with `x-json-tag` which replaces only the JSON part.

The generated model is the same.